python/lsst/ts/scheduler/driver/driver.py

```diff
@@ -233,13 +254,61 @@ def cold_start(self, observations):
         raise NotImplementedError("Cold start is not implemented.")
 
     def update_conditions(self):
+        """Update driver internal conditions.
+
+        When subclassing this method, make sure to call it at the start of the
+        method, as it performs operations like running the observatory through
+        the current targets on the queue.
         """
+        self.log.debug("Updating conditions.")
+        # Update observatory model with current observatory state.
+        self.models["observatory_model"].set_state(self.models["observatory_state"])
 
-        Returns
-        -------
+        self.models["sky"].update(self.models["observatory_state"].time)
 
-        """
-        pass
+        if self.is_night is None:
+            self.log.debug("Driver not initialized yet. Computing night parameters.")
+            # Driver was not initialized yet. Need to compute night
+            # boundaries
+
+            (self.current_sunset, self.current_sunrise) = self.models[
+                "sky"
+            ].get_night_boundaries(self.parameters.night_boundary)
+
+            self.is_night = (
+                self.current_sunset
+                <= self.models["observatory_state"].time
+                < self.current_sunrise
+            )
+
+            self.log.debug(
+                f"Sunset/Sunrise: {self.current_sunset}/{self.current_sunrise} "
+            )
+
+        is_night = self.is_night
+
+        self.is_night = (
+            self.current_sunset
+            <= self.models["observatory_state"].time
+            < self.current_sunrise
+        )
+
+        # Only compute night boundaries when we transition from nighttime to
+        # daytitme
+        if not self.is_night and is_night:
+            self.log.debug("Night over. Computing next nigth boundaries.")
+            self.night += 1
+            (self.current_sunset, self.current_sunrise) = self.models[
+                "sky"
+            ].get_night_boundaries(self.parameters.night_boundary)
+
+            self.log.debug(
+                f"[{self.night}]: Sunset/Sunrise: {self.current_sunset}/{self.current_sunrise} "
+            )
```

Comment:
Does this condition really only happen when the night is over (which I interpret as "at the end of the night") and not before the night starts (i.e. right after sunset but before the sun is at least 12 degrees below the horizon)?

Reply:
That is the idea.

For instance, 

```
self.is_night
```

will tell you if current time is between current sunset and sunrise (e.g. https://github.com/lsst-ts/ts_scheduler/blob/2944ebdf3026a7e4218adc4e4e8e0a7eca405e19/python/lsst/ts/scheduler/driver/driver.py#L290-L294).

So, the possibilities are:

1 - `self.is_night=True` and `is_night = True`: During the night (no need to compute anything).
2 - `self.is_night=False` and `is_night = True`: Transitioned from night/day (need to recompute night boundaries).
3 - `self.is_night=True` and `is_night = False`: Transitioned from day/night (no need to compute anything).
4 - `self.is_night=False` and `is_night = False`: During the day, no need to compute anything.

Reply:
I added a note on the code about that.